crates/tauri-utils/src/config/security.rs

```diff
@@ -0,0 +1,783 @@
+    match val {
+      HeaderSource::Inline(s) => {
+        let mut out: Vec<String> = Vec::new();
+        let mut separator = ',';
+        if s.contains(';') {
+          separator = ';';
+        }
+        for item in s.split(separator) {
+          out.push(String::from(item));
+        }
+        out
+      }
+      HeaderSource::List(l) => l,
+      HeaderSource::Map(m) => {
+        let mut out: Vec<String> = Vec::new();
+        for (key, value) in m.iter() {
+          let mut item = String::new();
+          item.push_str(key);
+          item.push(' ');
+          item.push_str(value);
+          out.push(item);
+        }
+        out
+      }
+    }
+  }
+}
+
+impl From<HeaderSource> for HashMap<String, String> {
+  fn from(val: HeaderSource) -> Self {
+    match val {
+      HeaderSource::Inline(s) => {
+        let mut out: HashMap<String, String> = HashMap::new();
+        for item in s.split(';') {
+          let index = item.find(' ');
+          if index.is_some() {
+            let (key, value) = item.split_at(index.unwrap());
+            out.insert(String::from(key), String::from(value));
+          }
+        }
+        out
+      }
+      HeaderSource::List(l) => {
+        let mut out: HashMap<String, String> = HashMap::new();
+        for item in l.iter() {
+          let index = item.find(' ');
+          if index.is_some() {
+            let (key, value) = item.split_at(index.unwrap());
+            out.insert(String::from(key), String::from(value));
+          }
+        }
+        out
+      }
+      HeaderSource::Map(m) => m,
+    }
+  }
+}
+
+impl From<String> for HeaderSource {
+  fn from(value: String) -> Self {
+    Self::Inline(value)
+  }
+}
+
+impl From<Vec<String>> for HeaderSource {
+  fn from(value: Vec<String>) -> Self {
+    Self::List(value)
+  }
+}
+
+impl From<HashMap<String, String>> for HeaderSource {
+  fn from(value: HashMap<String, String>) -> Self {
+    Self::Map(value)
+  }
+}
+
+/// A trait which implements on the [`Builder`] of the http create
+///
+/// Must add headers defined in the tauri configuration file to http responses
+pub trait HeaderAddition {
+  /// adds all headers defined on the config file, given the current HeaderConfig
+  fn add_configured_headers(self, headers: Option<&HeaderConfig>) -> Builder;
+}
+
+impl HeaderAddition for Builder {
+  /// Add the headers defined in the tauri configuration file to http responses
+  ///
+  /// this is a utility function, which is used in the same way as the `.header(..)` of the rust http library
+  fn add_configured_headers(mut self, headers: Option<&HeaderConfig>) -> Builder {
+    self = match headers {
+      Some(headers) => {
+        // Add the header Access-Control-Allow-Credentials, if we find a value for it
+        self = match &headers.access_control_allow_credentials {
+          Some(value) => self.header("Access-Control-Allow-Credentials", value.to_string()),
+          None => self,
+        };
```

Comment:
this can be simplified into
```suggestion
        if let Some(value) match &headers.access_control_allow_credentials {
          self = self.header("Access-Control-Allow-Credentials", value.to_string());
        };
```
and same should be done for other headers